internals/src/Tizen.Peripheral/Tizen.Peripheral/Uart.cs

```diff
@@ -0,0 +1,357 @@
+        }
+
+        /// <summary>
+        /// Closes the UART slave device.
+        /// </summary>
+        ~Uart()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Closes the UART slave device.
+        /// </summary>
+        public void Close() => Dispose();
+
+        /// <summary>
+        /// Disposes Uart object.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Disposes Uart object.
+        /// </summary>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            NativeUart.Close(_handle);
+            _disposed = true;
+        }
+
+        /// <summary>
+        /// Reads the bytes data from the UART slave device.
+        /// </summary>
+        /// <param name="buffer">The output byte array.</param>
+        /// <param name="offset">The offset in buffer at which to write the bytes.</param>
+        /// <param name="count">The number of bytes to read.</param>
+        /// <returns>The number of bytes read.</returns>
+        public int Read(byte[] buffer, int offset, int count)
+        {
+            if (count > buffer.Length - offset)
+                throw new Exception("Can not read more bytes than the buffer can hold.");
+            byte[] tmpBuffer = new byte[count];
+            var length = Convert.ToUInt32(count);
+            var ret = NativeUart.Read(_handle, tmpBuffer, length);
+            if (ret < Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+            Array.Copy(tmpBuffer, 0, buffer, offset, (int)ret);
+            return (int)ret;
+        }
+
+        /// <summary>
+        /// Writes the bytes data to the UART slave device.
+        /// </summary>
+        /// <param name="buffer">The byte array that contains the data to write to the device.</param>
+        /// <param name="offset">The offset in buffer at which to begin copying bytes.</param>
+        /// <param name="count">The number of bytes to write</param>
+        public int Write(byte[] buffer, int offset, int count)
+        {
+            if (count > buffer.Length - offset)
+                throw new Exception("Can not write more bytes than the buffer holds.");
+            byte[] tmpBuffer = new byte[count];
+            Array.Copy(buffer, offset, tmpBuffer, 0, count);
+            var length = Convert.ToUInt32(count);
+            var ret = NativeUart.Write(_handle, tmpBuffer, length);
+            if (ret < Internals.Errors.ErrorCode.None)
+                throw ExceptionFactory.CreateException(ret);
+            return (int)ret;
+        }
+
+        /// <summary>
+        /// Sets or gets baud rate of the UART slave device.
+        /// </summary>
+        public BaudRate BaudRate
+        {
+            get => _baudRate;
```

Comment:
All property has uninitialized value before set value, is it ok?

Reply:
This is ok, as native api also provides only "setters".
I added information about that to property documentation.
Properties are initialized after successful Set call.